### PR TITLE
Name changes

### PR DESCRIPTION
**Pull request by dANW34V3R**
This PR updates the names of many variables for code clarity and consistency which are listed below. Some of the updates still need to permeate the docs, but this will happen once the proposed names are approved.

- `operands `-> `sourceValues_` to prevent conflict with metadata.operands which contains both source and destination registers
-`operandsPending` -> `sourceOperandsPending_`
- `getStackPointer` -> `getInitialProcessStackPointer` this name was chosen as to not conflict with Linux.getInitialStackPointer for human readability
- Underscores added throughout architecture specific header files

### CODE REVIEW

**Review by ABenC377 — DISMISSED** [review]

File: t
Comment (JosephMoore25) [inline t:1]: Guessing this is an accident? I have no clue what this is, and even if it were correct, why `t`?

**Review by JosephMoore25 — CHANGES_REQUESTED** [review]
Simple PR, everything looks in order with what we agreed on.

Looks like the rogue file `t` which is a buggy copy of some regression tests made it in though, so this needs to be removed before approval.

**Review by JosephMoore25 — APPROVED** [review]

File: src/include/simeng/arch/aarch64/Instruction.hh
Comment (jj16791) [inline src/include/simeng/arch/aarch64/Instruction.hh:348]: Re-remove sourceRegisterCount_ that's been accidentally added back in

File: src/include/simeng/arch/aarch64/Instruction.hh
Comment (jj16791) [inline src/include/simeng/arch/aarch64/Instruction.hh:351]: Add newline back in

File: src/include/simeng/arch/riscv/Instruction.hh
Comment (jj16791) [inline src/include/simeng/arch/riscv/Instruction.hh:172]: Seem like Private member region already defined above unless GitHub is being odd

File: test/regression/riscv/RISCVRegressionTest.hh
Comment (jj16791) [inline test/regression/riscv/RISCVRegressionTest.hh:7]: Motivation for `constexpr` use? Not sure what compile-time advantage we can get here. Also, do you need the `const` specifier with a `constexpr` specifier?

**Review by jj16791 — CHANGES_REQUESTED** [review]

File: test/regression/riscv/RISCVRegressionTest.hh
Comment (dANW34V3R) [reply test/regression/riscv/RISCVRegressionTest.hh:7]: Was playing around with quashing warnings with @tom91136 yesterday. Shouldn't have been added to this PR so will remove

File: src/include/simeng/arch/riscv/Instruction.hh
Comment (dANW34V3R) [reply src/include/simeng/arch/riscv/Instruction.hh:172]: Good spot, removed

**Review by jj16791 — APPROVED** [review]

File: src/include/simeng/kernel/LinuxProcess.hh
Comment (FinnWilkinson) [inline src/include/simeng/kernel/LinuxProcess.hh:98]: Couldn't this be `getInitialStackPointer()`?

**Review by FinnWilkinson — CHANGES_REQUESTED** [review]

File: src/include/simeng/kernel/LinuxProcess.hh
Comment (FinnWilkinson) [reply src/include/simeng/kernel/LinuxProcess.hh:98]: Given they have to be called either by `linux_.getInitianStackPointer` or `process_.getInitialStackPointer` I think it will be evident what it refers to

Comment (dANW34V3R) [reply src/include/simeng/kernel/LinuxProcess.hh:98]: Will the referring to the same value change with multithreading?

Comment (FinnWilkinson) [reply src/include/simeng/kernel/LinuxProcess.hh:98]: They should do yes, but if this changes then said change should be made in the relevant multi-threading PR

**Review by FinnWilkinson — APPROVED** [review]

**Review by jj16791 — APPROVED** [review]

**Review by FinnWilkinson — APPROVED** [review]

**Review by JosephMoore25 — APPROVED** [review]

**Review by ABenC377 — APPROVED** [review]

**Pull request merged**
